ledger/src/blockstore/blockstore_purge.rs

```diff
@@ -239,6 +282,35 @@ impl Blockstore {
         }
         delete_range_timer.stop();
 
+        if let Some(mut slot_meta) = slot_meta {
+            let child_slot = slot_meta.slot;
+            if child_slot != from_slot || child_slot != to_slot {
+                error!("Slot meta parent cleanup was requested for {}, but a range was specified {} {}", child_slot, from_slot, to_slot);
+                return Err(BlockstoreError::InvalidRangeForSlotMetaCleanup);
+            }
```

Comment:
We don't have to do this sanity checking if we fetch the `SlotMeta` in this function ourselves instead of relying on the caller 

Technically, this would add some overhead since we'd be fetching the `SlotMeta` twice, once here and once in `Blockstore::clear_unconfirmed_slot()`. I think that's ok, but if we're conscious about that, I think we could remove the `SlotMeta` check in `Blockstore::clear_unconfirmed_slot()` anyways. The rocksdb range deletes are cheap, and this function should only be getting called if `ReplayStage` knows it has something that it needs to delete